tox.ini

```diff
@@ -29,6 +29,7 @@ deps =
     pytest-timeout
     pytest-openfiles
     pytest-flakefinder
+    parameterized
```

Comment:
why do we need this new dependency?

Reply:
i use old school unittest right now in some of my testing,  which doesnt interact well with pytest paramatrize... so this allows us to use the similar pattern.

I suspect I will convert it to pytest eventually (drop the unittest TestCase), but for me it was easier to reason through a class based test object rather than a pytest fixture.